course/auth.py

```diff
@@ -140,13 +152,20 @@ def __call__(self, request):
                 if request.user.is_superuser:
                     may_impersonate = True
                 else:
-                    qset = get_impersonable_user_qset(cast(User, request.user))
+                    if cur_course is not None:
+                        qset = get_impersonable_user_qset(cast(User, request.user),
+                                course_identifier=cur_course)
+                    else:
+                        qset = get_impersonable_user_qset(cast(User, request.user),
+                                course_identifier=imp_course)
```

Comment:
If the impersonation is course-scoped, no impersonation should happen if `cur_course` is None.

Reply:
I'll have to selectively allow `/stop_impersonating` (or change `stop_impersonating` url to also be course based) and signout page, right?

Reply:
No, I don't think that's necessary. `/stop_impersonating` will look directly at the values in the session and will work OK even if the `request.user` is not the impersonee.

Reply:
Same for sign-out.

Reply:
I'll have to stop the error message though, otherwise clicking that button would give me,

![image](https://user-images.githubusercontent.com/5234427/64226172-e6484d80-cea3-11e9-80eb-26a7c583c12e.png)

Reply:
Sure. But if they're already impersonating someone, it's fine to assume that they were (at some point) permitted to do that.